src/player.rs

```diff
@@ -11,6 +13,18 @@ use metadata::{FileFormat, Track, TrackRef};
 use session::{Bitrate, Session};
 use util::{self, ReadSeek, SpotifyId, Subfile};
 pub use spirc::PlayStatus;
+use std::sync::atomic::{AtomicUsize, Ordering};
+use std::process::exit;
+
+use nix::sys::signal;
+
+static mut received_signal: AtomicUsize = AtomicUsize::new(0);
```

Comment:
You can use `ATOMIC_USIZE_INIT`, which would remove the need for `const_fn`.
Also why a usize rather a bool ?